wagtail/wagtailsearch/backends/elasticsearch.py

```diff
@@ -398,6 +399,38 @@ def __repr__(self):
 
 
 class ElasticsearchSearchResults(BaseSearchResults):
+    def __init__(self, *args, **kwargs):
+        super(ElasticsearchSearchResults, self).__init__(*args, **kwargs)
+        self.highlight_params = {}
+
+    def highlight(self, **kwargs):
+        # It's possible to pass list of dicts and dict as the fields param
+        fields = kwargs.get('fields', [])
+        if isinstance(fields, dict):
+            fields = [{k: v} for k, v in fields.items()]
+
+        post_processed_fields = {}
+        for field_def in fields:
+            field_name = field_def.keys()[0] if isinstance(field_def, dict) and len(field_def) else None
+            if field_name:
+                # Use wildcard column name on request. This allows us to query against all content types.
+                # Alternative solution: get all indexed models and add all possible column names into request.
+                field_column_name = '*{}'.format(field_name)
+                post_processed_field_def = {
+                    field_column_name: field_def[field_name]
+                }
+                post_processed_fields[field_name] = post_processed_field_def
+
+        self.highlight_params['fields'] = post_processed_fields
+        self.highlight_params['require_field_match'] = kwargs.get('require_field_match', None)
```

Comment:
Thanks for the expanation! If I understood correctly, it sounds like it always needs to be ``false`` in order for highlighting to work as expected, so does it need to be configurable?